homeassistant/components/powerwall/sensor.py

```diff
@@ -32,17 +36,22 @@
 _METER_DIRECTION_EXPORT = "export"
 _METER_DIRECTION_IMPORT = "import"
 
+_ValueParamT = TypeVar("_ValueParamT")
+_ValueT = TypeVar("_ValueT", bound=StateType | date | datetime | Decimal)
```

Comment:
I think this type can be narrowed as the powerwall integration doesn't use all of these types

Reply:
I was just following the [typing information](https://github.com/home-assistant/core/blob/dev/homeassistant/components/sensor/__init__.py#L208) for what any sensor entity could possibly return.   Other integrations use the same typing hints.   Should I reduce the list to what is actually used?

Reply:
Yes please narrow the type to match only what is needed.